<h1 align="center">Investigate Stroop Effect</h1>
***


<font size='4'>In this project will review the basic concepts of statistics, including:
 - How to identify components of an experiment.
 - How to use descriptive> statistics to describe qualities of a sample.
 - How to set up a hypothesis test, make inferences from a sample, and draw conclusions based on the results.


<font size='4'>What is our independent and dependent variable?<br> 
>Independent variable is a variable that is being manipulated in an experiment in order to observe the effect
on a dependent variable.It is usually plotted along the x-axis of a graph.<br>
Congruent/Incogruent Stroop Test Score is the independent variable and is plotted on the X-axis.<br>
The time taken to perform Congruent/Incogruent test in seconds is the dependant variable and is usually
plotted on Y axis.

<font size='4'>What is an appropriate set of hypotheses for this task? What kind of statistical test do you expect to perform? Justify your choices.<br>
>Null Hypothesis:The mean of the population congruntent and Incorgruent stroop test are equal.<br>
$H_o : \mu_c=\mu_i$<br>
>Alternative hypothesis:The mean of population congruent test and incrocruent test are not equal.Since the popuation standard deviation is not known we will do t-Test for paired sample which are dependant.The sample is dependent because we are measuring the time taken to complete the stroop test for congruent and incogruent data but for the same person.<br>
$H_a : \mu_c \ne \mu_i$<br>
We will measure effect size by calculating Cohen D and calculate the Confidence interval of the difference in times(in second). between the two test.

<font size=4>Assumptions regarding the paired t-Test are as follows:

<font size='4'>Interval or ratio scale of measurement (approximately interval)<br>
 - Random sampling from a defined population.
 - Samples or sets of data used to produce the difference scores are linked in the population through repeated   
   measurement,natural association, or matching.
 - Scores are normally distributed in the population; difference scores are normally distributed.

<font size='4'> Report some descriptive statistics regarding this dataset.Include at least one measure of central tendency and at least one measure of variability.?

In [4]:
#importing pandas
import pandas as pd

In [5]:
# Reading the csv file
df=pd.read_csv('./stroopdata.csv')

<font size='4'>We calculate the mean,median and mode of congruent and incongruent test timing in sec.<br><br>
Table below we calculate the following Count:The number of data sample in each category.<br><br>
Mean: Mean in sec<br>
std: Standard deviation<br>
min: least value in sec in each category.<br>
25%: 25% value in sec 1 IQR<br>
50%: Median value in sec in each category.<br>
75%: 75% value in sec 3 IQR.<br>
max: Mode value in sec in each category. 4 IQR<br>

In [6]:
df.describe()

Congruent  Incongruent
count  24.000000    24.000000
mean   14.051125    22.015917
std     3.559358     4.797057
min     8.630000    15.687000
25%    11.895250    18.716750
50%    14.356500    21.017500
75%    16.200750    24.051500
max    22.328000    35.255000

<font size='4'>Provide one or two visualizations that show the distribution of the sample data. Write one or two sentences noting what you
observe about the plot or plots.?

In [7]:
# Bar char to be done

<font size='4'>Bar Chart clearly show that time take to complete Incongruent stroop test is greater than the time taken to
complete Congruent stroop test in all the sample data.

In [8]:
#Box or wishker plot to

Box plot show the spread of data in the time(sec) taken to complete the
Congruent/Incogruent Stroop Test
###Range congruent = max‐min=13.698
###Range incongruent = max‐min=19.568
CONGRUENT INCONCRUENT
min 8.63 15.68
25% 11.52 18.66
50% 14.23 21.01
75% 16.59 24.36
max 22.32 35.25

<font size='4'>Now, perform the statistical test and report your results. What is your confidence level and your critical
statistic value? Do you reject the null hypothesis or fail to reject it? Come to a conclusion in terms of the experiment task. Did the results match up with your expectations?

In [ ]:
#calculation

#References
http://www.imathas.com/stattools/boxplot.html
(http://www.imathas.com/stattools/boxplot.html) used to create
the box plot.
udacity notes and excel to calculate the t stastistic
https://www.khanacademy.org/math/probability/desc

https://www.khanacademy.org/math/probability/descriptivestatistics/
boxandwhiskerplots/
v/rangeandmidrange
(https://www.khanacademy.org/math/probability/descriptivestatistics/
boxandwhiskerplots/
v/rangeandmidrange)
: how to
read box plot
http://www.statstutor.ac.uk/resources/uploaded/pairedttest.
pdf
(http://www.statstutor.ac.uk/resources/uploaded/pairedttest.
pdf)
t
test for dependant
http://www.psychology.emory.edu/clinical/bliwise/Tutorials/TOM/meanstests/(http://www.psychology.emory.edu/clinical/bliwise/Tutorials/TOM/meanstests/for
assumptions on T test